students/ZackConnaughton/lessons/lesson07/assignment/database.py

```diff
@@ -0,0 +1,234 @@
+"""
+creates database with customer info and product info
+"""
+
+from threading import Thread
+import csv
+import queue as Queue
+from pymongo import MongoClient
+import time
+
+
+
+class MongoDBConnection():
+    """MongoDB Connection"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+def show_available_products():
+    mongo = MongoDBConnection()
+
+    with mongo:
+        database = mongo.connection.media
+
+        output_dict = list(database.products.find({'quantity_available': {'$gt':'0'}}, {"_id": 0}))
+
+        print(output_dict)
+        return(output_dict)
```

Comment:
Once you have your output_dict, you should close out of the context manager:
```
with mongo:
    database = mongo.connection.media
    output_dict = list(database.products.find({'quantity_available': {'$gt':'0'}}, {"_id": 0}))
print(output_dict)
return output_dict
```
True for other functions as well. The unindent closes the connection.